packages/fxa-auth-server/lib/payments/subscription-reminders.ts

```diff
@@ -0,0 +1,216 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Logger } from 'mozlog';
+import Stripe from 'stripe';
+import { Container } from 'typedi';
+
+import { ConfigType } from '../../config';
+import { reportSentryError } from '../sentry';
+import { SentEmailParams, Plan } from 'fxa-shared/subscriptions/types';
+import { StripeHelper, TimeSpanInS } from './stripe';
+import { SentEmail } from 'fxa-shared/db/models/auth';
+
+import {
+  DEFAULT_PLAN_LENGTH,
+  DEFAULT_REMINDER_LENGTH,
+} from '../../scripts/subscription-reminders';
+
+export const EMAIL_TYPE: string = 'subscriptionRenewalReminder';
+const DAYS_IN_A_WEEK: number = 7;
+const DAYS_IN_A_MONTH: number = 30;
+const DAYS_IN_A_YEAR: number = 365;
+export const MS_IN_A_DAY: number = 24 * 60 * 60 * 1000;
+
+interface EligiblePlansByInterval {
+  day(plan: Plan, planLength: number): Boolean;
+  week(plan: Plan, planLength: number): Boolean;
+  month(plan: Plan, planLength: number): Boolean;
+  year(plan: Plan, planLength: number): Boolean;
+}
+
+const eligiblePlansByInterval: EligiblePlansByInterval = {
+  day: (plan, planLength) => plan.interval_count >= planLength,
+  week: (plan, planLength) =>
+    plan.interval_count >= planLength / DAYS_IN_A_WEEK,
+  month: (plan, planLength) =>
+    plan.interval_count >= planLength / DAYS_IN_A_MONTH,
+  year: (plan, planLength) =>
+    plan.interval_count >= planLength / DAYS_IN_A_YEAR,
+};
+
+export class SubscriptionReminders {
+  private db: any;
+  private mailer: any;
+  private stripeHelper: StripeHelper;
+
+  constructor(
+    private log: Logger,
+    config: ConfigType,
+    private planLength: number = DEFAULT_PLAN_LENGTH,
+    private reminderLength: number = DEFAULT_REMINDER_LENGTH,
+    db: any,
+    mailer: any
+  ) {
+    this.db = db;
+    this.mailer = mailer;
+    this.stripeHelper = Container.get(StripeHelper);
```

Comment:
Even when there are default values for the plan length and reminder length, I think you'd still need assign them to the object's properties.

Reply:
Why not accept the StripeHelper through a parameter?  That why you won't have a mixture of argument values and DI container values.